### PR TITLE
Remove common-criteria for all patterns migration test

### PR DESCRIPTION
**Pull request by lemon-suse**
We need ensure patterns-certification-common-criteria pattern is NOT installed in the "all patterns" case (because it causes a change in default behaviour). For migration test, we need remove the pattern if it is installed and don't install it if it is not installed. 

- Related ticket: https://progress.opensuse.org/issues/73645
- Needles: N/A
- Verification run: verification on OSD
                          - offline migration:  https://openqa.nue.suse.com/tests/4875791
                          - online migration:  https://openqa.nue.suse.com/tests/4875756#

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -1729,4 +1730,98 @@ sub common_service_status {
...
+        @pt_list_un = get_pattern_list "zypper pt -u", "|";
+    }
+    # install certain pattern from parameter.
+    else {
+        @pt_list_un = split(/,/, get_var('PATTERNS'));
+    }
+
+    my %installed_pt = ();
+    foreach (@pt_list_in) {
+        if ($_ =~ /common-criteria/) {
```
Comment (coolgw) [inline lib/utils.pm:1797]: comments here why we need this remove action

File: lib/utils.pm [hunk 2]
```diff
@@ -1729,4 +1730,107 @@ sub common_service_status {
...
+    my @pt_list_un;
+    my @pt_list_in;
+
+    @pt_list_in = get_pattern_list "zypper pt -i", "i";
+    # install all patterns from product.
+    if (check_var('PATTERNS', 'all')) {
+        @pt_list_un = get_pattern_list "zypper pt -u", "|";
+        my $index = 0;
+        foreach (@pt_list_un) {
+            if ($_ =~ /common-criteria/) {
```
[the remark this answers is not in the document]
Comment (lemon-suse) [reply lib/utils.pm]: This is outdated code.

File: lib/utils.pm (hunk 1 above)
Comment (lemon-suse) [reply lib/utils.pm:1798]: Added comments.

**Comment by coolgw** [comment]
LGTM

**Pull request merged**